src/main/java/io/appium/java_client/service/local/AppiumDriverLocalService.java

```diff
@@ -219,4 +232,127 @@ public void addOutPutStreams(List<OutputStream> outputStreams) {
     public boolean clearOutPutStreams() {
         return stream.clear();
     }
+
+    /**
+     * Enables server output data logging through
+     * <a href="http://slf4j.org">SLF4J</a> loggers. This allow server output
+     * data to be configured with your preferred logging frameworks (e.g.
+     * java.util.logging, logback, log4j).</br>
+     * </br>
+     * NOTE1: You might want to call method {@link #clearOutPutStreams()} before
+     * calling this method. </br>
+     * NOTE2: it is required that {@link --log-timestamp} server flag is
+     * {@link false}. </br>
+     * </br>
+     * By default log messages are:
+     * <li>logged at {@code INFO} level, unless log message is pre-fixed by
+     * {@code [debug]} then logged at {@code DEBUG} level.</li>
+     * <li>logged by a <a href="http://slf4j.org">SLF4J</a> logger instance with
+     * a name corresponding to the appium sub module as prefixed in log message
+     * (logger name is transformed to lower case, no spaces and prefixed with
+     * "appium.service.").</li> </br>
+     * Example log-message: "[ADB] Cannot read version codes of " is logged by
+     * logger: {@code appium.service.adb} at level {@code INFO}. </br>
+     * Example log-message: "[debug] [XCUITest] Xcode version set to 'x.y.z' "
+     * is logged by logger {@code appium.service.xcuitest} at level
+     * {@code DEBUG}.
+     * 
+     * </br>
+     * </br>
+     * 
+     * @see #addSlf4jLogMessageConsumer(BiConsumer)
+     */
+    public void enableDefaultSlf4jLoggingOfOutputData() {
+        addSlf4jLogMessageConsumer((logMessage, ctx) -> {
+            if (ctx.level().equals(DEBUG)) {
```

Comment:
This mapping is not sufficient, since Appium server uses more levels, for example `warn` and `error`

Reply:
Will add the other levels as well.

Reply:
Hmm, I have never seen an actual log message on either `warn`/`error` level. Do you know if they follow same format as `debug`? i.e. log message is prefixed with log level. `Info` level tag is not added in log message.

Reply:
I was just able to trigger a `WARN`message, but there is no log level indication in the log message it self. So will ignore `warn` and `error` levels for now.

Example
```
[Appium] Deprecated server args: 
[Appium]   --device-name => --default-capabilities '{"deviceName":"someName"}'
```
According to this, the above log message is at `WARN`level:
https://github.com/appium/appium/blob/7a00b3f875991ba1d95359ed53a3eda47911f64b/lib/main.js#L44

Reply:
the actual level is added there if logging timestamp is enabled:

```
2018-08-31 08:17:13:813 - info: [debug] [JSONWP Proxy] Got response with status 200: {"value":{"using":"class chain","value":"**/XCUIElementTypeCollectionView[`name == 'add_participants.list'`]/XCUIElementTypeCell[$name == 'user_cell.name' AND value == 'qrj7pcz5'$]","description":"unable to find an element"},"sessionId":"A4FBAD14-ED57-46D4-9407-B91C76BCB415","status":7}
2018-08-31 08:17:13:814 - info: [debug] [MJSONWP] Matched JSONWP error code 7 to NoSuchElementError
2018-08-31 08:17:13:815 - error: [W3C] Encountered internal error running command: NoSuchElementError: An element could not be located on the page using the given search parameters.
2018-08-31 08:17:13:815 - error: [W3C]     at XCUITestDriver.callee$0$0$ (/usr/local/lib/node_modules/appium/node_modules/appium-xcuitest-driver/lib/commands/find.js:130:13)
```